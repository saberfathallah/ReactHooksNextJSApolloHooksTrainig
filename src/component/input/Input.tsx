import TextField from "@material-ui/core/TextField";

interface InputProps {
  handleChange: any;
  handleBlur: any;
  values: {
    description: string;
  },
  label: string;
  variables: any;
  errors: {
    description: string;
  }
  touched: any
  style: any
}

const Input = (props: InputProps) => {
  const {
    handleChange,
    handleBlur,
    values,
    label,
    variables,
    errors,
    touched,
    style
  } = props;

  return (
    <>
      <TextField
        type="description"
        name="description"
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.description}
        id={`${variables.description}standard-full-width`}
        label={label}
        style={{ ...style }}
        placeholder=""
        fullWidth
        margin="normal"
        InputLabelProps={{
          shrink: true,
        }}
      />
      {errors.description && touched.description && errors.description}
    </>
  );
};

export default Input;
