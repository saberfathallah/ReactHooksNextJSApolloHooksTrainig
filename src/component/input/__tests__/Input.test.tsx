import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Input from "../Input";

describe("Input", () => {

  const props = {
    label: "Ajouter un commentaire",
    handleChange: () => {},
    handleBlur: () => {},
    values: { description :''},
    errors: { description :''},
    touched: { description :''},
    style: { margin: 8 },
    variables: {
      description: "description",
    },
  };

  it("should display snapshot with Ajouter un commentaire", () => {
    const tree = renderer.create(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display snapshot with Modifier ton commentaire", () => {
    props.label = "Modifier ton commentaire"
    const tree = renderer.create(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display Modifier ton commentaire", () => {
    const wrapper = mount(<Input {...props} />);
    expect(wrapper.text()).toContain("Modifier ton commentaire");
  });
});