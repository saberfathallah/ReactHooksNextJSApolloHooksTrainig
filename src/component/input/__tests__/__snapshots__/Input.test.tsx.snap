// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should display snapshot with Ajouter un commentaire 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
      htmlFor="descriptionstandard-full-width"
      id="descriptionstandard-full-width-label"
    >
      Ajouter un commentaire
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="descriptionstandard-full-width"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={false}
        type="description"
        value=""
      />
    </div>
  </div>,
  "",
]
`;

exports[`Input should display snapshot with Modifier ton commentaire 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
      htmlFor="descriptionstandard-full-width"
      id="descriptionstandard-full-width-label"
    >
      Modifier ton commentaire
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="descriptionstandard-full-width"
        name="description"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={false}
        type="description"
        value=""
      />
    </div>
  </div>,
  "",
]
`;
